janusgraph-core/src/main/java/org/janusgraph/graphdb/query/index/IndexSelectionUtil.java

```diff
@@ -25,4 +40,29 @@ public static boolean indexCoversOrder(MixedIndexType index, OrderList orders) {
         }
         return true;
     }
+
+    public static Set<IndexType> getMatchingIndexes(MultiCondition<JanusGraphElement> conditions) {
```

Comment:
Nitpick: The result of this method is always filtered. Instead of filling `Set` with values and then removing them, we could just pass filter (or predicate) function to this method and filter it directly here without conducting additional pass to filter the resulting set afterwards.

Reply:
I see what you mean. Do you think it's worth creating an issue to rewrite the method?

Reply:
I think issue could be helpful so that we don't forget to change it later

Reply:
Created #2339